### ISSUE TITLE
Integrate a web music player?

### ISSUE THREAD

**Issue by dashdsrdash** [owntone/owntone-server#1341]
This is an enhancement request for the web portion to include a music player pointed at /stream.mp3 and obeying / generating the normal controls.

This won't work for background playing on all devices/browsers, but will certainly work for the cases where an Owntone user wants to listen on the same device as they are using to control Owntone -- Android phones using Firefox will run in the background, Mac OS devices will run webpages in the background, other cases are likely.

Something as simple as this might work:

```
 <audio controls="true" preload="none">
 <source src="https://owntone.local:3689/stream.mp3" type="audio/mp3">Your browser does not support the audio element.
 </audio>
```

**Comment by ejurgensen** [comment]
Perhaps I misunderstand, but isn't that already available? In the speaker list you can select HTTP STREAM (stream.mp3).

**Comment by dashdsrdash** [comment]
ejurgensen wrote: 
> Perhaps I misunderstand, but isn't that already available? In the speaker list you can select HTTP STREAM (stream.mp3).

That pops open a new window on almost all of my available
browsers; in some of them, it immediately fails because it can't
find a video stream to go with it.

By specifying an audio element instead of an href, I hope to
solve both problems at once.

**Comment by ejurgensen** [comment]
Sounds peculiar that a browser would expect video for an mp3?

In any case, OwnTone's frontend contributor is inactive, and I don't know much about this stuff. I recall there was some discussion here on github of what the most cross-browser compatible way of doing this was. I couldn't find it right now, but if you come across it please add a mention of it here.

**Comment by dashdsrdash** [comment]
ejurgensen wrote: 
> Sounds peculiar that a browser would expect video for an mp3?
> 
> In any case, OwnTone's frontend contributor is inactive, and I don't know much about this stuff. I recall there was some discussion here on github of what the most cross-browser compatible way of doing this was. I couldn't find it right now, but if you come across it please add a mention of it here.
> 

OK, if I can come up with a good patch it might be welcome?

**Comment by whatdoineed2do** [comment]
@dashdsrdash 
Theres only 2 places that this stream link is available and both in this file:

https://github.com/owntone/owntone-server/blob/4c5217d3d93b198f04bfbef1a2ea71cb3033edb0/web-src/src/components/NavbarBottom.vue#L84

Wrapping with the `audio` tag should be simple enough but does need a rebuild of the vuejs stuff though

**Comment by ejurgensen** [comment]
It seems chme didn't go with the audio tag [due to some issues](https://github.com/owntone/owntone-server/issues/621#issuecomment-462649758), but I'm not sure what they were. I'm reluctant to make any change here, due to missing the frontend maintainer.

**Comment by Joshfindit** [comment]
Works well enough in FireFox and Brave for me (a bit of delay that is probably the fault of the browser’s audio buffer, but sound plays fine).

Common browser-based media issues are:

- Header mismatch from the server (fine here: reported as `audio/mpeg`)
- Bad metadata for the link (also fine here: simple `a href=“stream.mp3”`)
- Adding “download” to the href (not the issue: not in the link)
- Adding `attachment` to the server response (not an issue, but could be optimized as there’s nothing in `Content-Disposition` but it could be specified as `inline`)

**Comment by Joshfindit** [comment]
Also, it’s been ages by HTML standards: the `audio` tag may now work out of the box.

Not sure if this is in another issue, but clicking the icon for HTTP STREAM actually does play it in-browser, and with little or no delay (in my 20 seconds of testing)

**Comment by chme** [comment]
The intention for the stream.mp3 link is to give users the possibility to easily copy the link if they want to open it in a different player.

The click on the icon for the http stream starts the playback in the browser by using the audio js api (intention was to have a similar behavior as we have for e. g. airplay speaker). 

There is a bug in web interface that results in rendering the icon behind an invisible div on mobile phones / tablets. This makes the icon barely clickable (only the upper right corner is working).

I fixed this in #1374.

**Comment by dashdsrdash** [comment]
Christian Meffert wrote: 
> The click on the icon for the http stream starts the playback in the browser by using the audio js api (intention was to have a similar behavior as we have for e. g. airplay speaker). 
> 
> There is a bug in web interface that results in rendering the icon behind an invisible div on mobile phones / tablets. This makes the icon barely clickable (only the upper right corner is working).
> 
> I fixed this in #1374.

Thanks very much! This wasn't obvious to me, and I confirm that
it works on a desktop browser and (pre-patching) requires a zoom
and a lot of random tapping attempts to work on Firefox for
Android.
